### PR TITLE
Correctly apply styles to email links

### PR DESCRIPTION
**Pull request by Gonals**
cc @stitesExpensify 

There's a bit of duplicate code, but I thought this was better than an if/else where link and email handling share the code. I can be convinced otherwise, though

### Fixed Issues
$ https://github.com/Expensify/App/issues/13423

# Tests
1. Log into your account in NewDot and open a chat
2. Send `[my *web*](www.google.es)`
3. Confirm the word `web` is bold
4. Send `[my *email*](alberto@expensify.com)`
5. Confirm `email` is bold
6. Send `*[my email](alberto@expensify.com)*`
7. Confirm `my email` is bold

# QA
Same as tests

### CODE REVIEW

**Comment by Gonals** [comment]
The issue seems to be already fixed

**Comment by Gonals** [comment]
- - -
   ***I have read the CLA Document and I hereby sign the CLA***
   - - -

File: lib/ExpensiMark.js [hunk 1]
```diff
@@ -55,7 +55,13 @@ export default class ExpensiMark {
              */
             {
                 name: 'email',
-                regex: /\[([^[\]]*)\]\(([a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z]+?(\.[a-zA-Z]+)+)\)/gim,
+                process: (textToProcess, replacement) => {
+                    const regex = new RegExp(
+                        '\\[([^[\\]]*)]\\(([a-zA-Z0-9.!#$%&\'*+\\/=?^_`{|}~-]+@[a-zA-Z]+?(\\.[a-zA-Z]+)+)\\)', 'gim'
```
Comment (amyevans) [inline lib/ExpensiMark.js:60]: This is pretty hard to parse, and when I put it in regex101.com I get a pattern error:
Screenshot 2022-12-09 at 5 26 29 PM

a.) Just curious if you have a different tool you were using to build it/explain it
b.) Maybe we can add some comments or break it up into descriptive variable pieces a bit to make it somewhat more human-readable?

**Review by amyevans — COMMENTED** [review]
Tests well but a question for you since Regex is Hard 😄

File: lib/ExpensiMark.js (hunk 1 above)
Comment (Gonals) [reply lib/ExpensiMark.js:60]: Oh, I'm reusing the existing regex, but creating a new Regex required escaping some characters we didn't have to escape before (all those double `\\`). If you make them just one `\`, regex101 will give you all the info:

Screenshot 2022-12-12 at 10 55 15 AM

**Comment by Gonals** [comment]
Comments addressed!

File: lib/ExpensiMark.js (hunk 1 above)
Comment (amyevans) [reply lib/ExpensiMark.js:60]: Okay I see, thanks!

A couple more thoughts:
1. What do you think about breaking it up for legibility like so:
```js
const firstPartOfEmail = '[a-zA-Z0-9.!#$%&\'*+\\/=?^_`{|}~-]+';
const secondPartOfEmail = '[a-zA-Z]';
const thirdPartOfEmail = '[a-zA-Z]';
const regex = new RegExp(
    `\\[([^[\\]]*)]\\((${firstPartOfEmail}@${secondPartOfEmail}+?(\\.${thirdPartOfEmail}+)+)\\)`, 'gim'
);
```

2. This regex is not allowing for numbers or hyphens in the domain name of the email address, e.g. it doesn't match `alberto@expensify22.com` or `alberto@expens-ify.com`. Perhaps that's outside the scope of this issue though?

**Review by amyevans — APPROVED** [review]
Tossing it over to you @BStitesExpensify

**Review by stitesExpensify — APPROVED** [review]
LGTM and tests well!

**Comment by stitesExpensify** [comment]
recheck

**Comment by stitesExpensify** [comment]
Hm IDK why that didn't work (or why your signing of the CLA didn't work). Maybe try again without the bars? Like just do `I have read the CLA Document and I hereby sign the CLA` by itself?

**Comment by Gonals** [comment]
- - -
   ***I have read the CLA Document and I hereby sign the CLA***
   - - -

**Comment by Gonals** [comment]
That did it!
ready for merge @amyevans, @stitesExpensify
